### PR TITLE
Add sorting of connectors properties based on order from API

### LINKED ISSUE #2042: Most (all?) Connectors are not displaying links to docs in setup pages
## Expected Behavior
* When setting up a connector, I should see a link to the docs for that specific connector.
Screen Shot 2021-05-26 at 10 58 08 AM
* While if the documentation url isn't available this link should not appear, in cases where it is available it should (and it seems like there are a lot of cases right now where we have this documentation url but it is not appearing).

## Current Behavior
* After sampling a bunch of connectors none of them have this link anymore. Not sure if this is a FE thing where this got removed by accident or a configuration problem for many connectors.

Screen Shot 2021-02-11 at 9 21 23 AM

In addition it would be nice if this link appeared in the settings page as well: https://www.figma.com/file/etaOkOsnXMoKwbfhCD71kw/Web-app?node-id=21501%3A23 . Wherever it is convenient to fit it in.

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1200367912513076/1200368089356840) by [Unito](https://www.unito.io)

### LINKED ISSUE #2998: UI should respect `order` metadata in spec configs
## Tell us about the problem you're trying to solve
In some specs we add an `order` field as metadata (e.g. [postgres spec](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-postgres/src/main/resources/spec.json)). The idea is that this order should be the order in which the fields are displayed in the UI. 

## Describe the solution you’d like
* If present the UI should respect the `order` metadata if present.
* If no `order` is present, then behavior should remain the same.
* If some fields have order and some do not those without order go to the back sorted alphabetically (or whatever is easiest to implement on the UI
* If multiple fields have the same `order` value, tie break alphabetically.

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1200367912513076/1200368161513809) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by jamakase**
## What
Closes #2998 
Closes #2042

### CODE REVIEW

File: airbyte-webapp/src/components/ServiceForm/components/FormSection.tsx [hunk 1]
```diff
@@ -129,12 +129,37 @@ const ArraySection: React.FC<{
   );
 };
 
+function OrderComparator(a: FormBlock, b: FormBlock): number {
```
Comment (cgardens) [inline airbyte-webapp/src/components/ServiceForm/components/FormSection.tsx:132]: we should fall back on sorting alphabetically if we don't have an order field. i'm not seeing that logic here. does that still need to be implemented it or am i being blind?

**Review by cgardens — APPROVED** [review]
one question, but once resolved feel free to merge.

File: airbyte-webapp/src/components/ServiceForm/components/FormSection.tsx (hunk 1 above)
Comment (Jamakase) [reply airbyte-webapp/src/components/ServiceForm/components/FormSection.tsx:132]: Yeah. We do check on line 156 and do not sort at all if no order present.

Comment (cgardens) [reply airbyte-webapp/src/components/ServiceForm/components/FormSection.tsx:132]: great!

**Pull request merged**